synapse/cortex.py

```diff
@@ -5898,15 +5900,17 @@ async def _getMirrorProxy(self, opts):
             logger.warning('Storm query mirror pool is empty, running query locally.')
             return None
 
+        timeout = self.stormpoolopts.get('timeout:connection')
+
         for _ in range(size):
 
             try:
-                timeout = self.stormpoolopts.get('timeout:connection')
                 proxy = await self.stormpool.proxy(timeout=timeout)
+
                 proxyname = proxy._ahainfo.get('name')
                 if proxyname is not None and proxyname == self.ahasvcname:
-                    # we are part of the pool and were selected. Convert to local use.
-                    return None
+                    # we are part of the pool and were selected. Skip.
```

Comment:
Should we just make it so the leader cannot be part of the query mirror pool since it doesn't do anything?